ghcide/src/Development/IDE/Core/ProgressReporting.hs

```diff
@@ -42,38 +48,60 @@ data ProgressEvent
   | ProgressCompleted
   | ProgressStarted
 
-data ProgressReporting m = ProgressReporting
-  { progressUpdate :: ProgressEvent -> m (),
-    inProgress     :: forall a. NormalizedFilePath -> m a -> m a,
-    -- ^ see Note [ProgressReporting API and InProgressState]
-    progressStop   :: IO ()
+data ProgressReportingNoTrace m = ProgressReportingNoTrace
+  { progressUpdateI :: ProgressEvent -> m (),
```

Comment:
You might not need to add name suffixes if you use `DuplicateRecordFields`?

Reply:
No, we have class function and data field collision. `DuplicateRecordFields ` is not helpful here

Reply:
But maybe we can make it prefix with '_' and not export it.